From d31ed4f961deff00ef5c2299ad1db8a2a705c5bc Mon Sep 17 00:00:00 2001
From: Juntang Fu(David) <juntang.fu@windriver.com>
Date: Thu, 11 Aug 2011 11:41:44 +0800
Subject: [PATCH] Add tracepoint to trace executed thread name changed by prctl

Currently the way of LTTng handles thread names is by adding a
userspace thread event. It is executed at thread startup. So LTTng
cannot know the name of threads already executing before we started
tracing. Now add a new tracepoint event to trace the executed thread
name.

Signed-off-by: David Fu <Juntang.Fu@windriver.com>
---
 fs/exec.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/fs/exec.c b/fs/exec.c
index f0915a4..587b92f 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1000,6 +1000,7 @@ void set_task_comm(struct task_struct *tsk, char *buf)
 	strlcpy(tsk->comm, buf, sizeof(tsk->comm));
 	task_unlock(tsk);
 	perf_event_comm(tsk);
+	trace_fs_exec(tsk->comm); 
 }
 
 int flush_old_exec(struct linux_binprm * bprm)
-- 
1.7.0.2

